File: katas/7 kyu/complementary-dna/solution.py
from ntpath import join
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def DNA_strand(dna) :
  return dna.translate(dna.maketrans('ATGC' , 'TACG'))
# ----------------------------------- TEST ----------------------------------- #
@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():     
        test.assert_equals(DNA_strand("AAAA"),"TTTT","String AAAA is")
        test.assert_equals(DNA_strand("ATTGC"),"TAACG","String ATTGC is")
        test.assert_equals(DNA_strand("GTAT"),"CATA","String GTAT is")

File: katas/7 kyu/complementary-dna/solution.php
<!-- complementary-dna -->
<?php
use PHPUnit\Framework\TestCase;
/* ------------------------------- MY SOLUTION ------------------------------ */
function DNA_strand($dna) {
  return strtr($dna , 'ATCG' , 'TAGC');
}
/* ---------------------------------- TEST ---------------------------------- */
class ComplementaryDNATest extends TestCase {
  public function testExamples() {
    $this->assertEquals("TTTT", DNA_strand("AAAA"));
    $this->assertEquals("AAAA", DNA_strand("TTTT"));
    $this->assertEquals("TAACG", DNA_strand("ATTGC"));
    $this->assertEquals("ATTGC", DNA_strand("TAACG"));
    $this->assertEquals("CATA", DNA_strand("GTAT"));
    $this->assertEquals("GTAT", DNA_strand("CATA"));
  }
}